pennylane_lightning/lightning_qubit.py

```diff
@@ -354,40 +408,61 @@ def apply_lightning(self, state, operations):
         Returns:
             array[complex]: the output state tensor
         """
-        state_vector = np.ravel(state)
+        batch_size_op = self.operations_batch_size(operations)
```

Comment:
Note that a `QuantumTape` comes with a pre-processed `batch_size` attribute. Can't we reuse that one, instead of implementing `operations_batch_size`? :)

Reply:
The apply method receives a list of operations to be applied, not a tape.

Reply:
Okay. I suppose as we do not recompute the `batch_size` but just retrieve it from the operations -- or from the first one that has a `batch_size!=None` -- it is not very expensive :)